docs/articles/documentation/guides/basic-guides/obtain-client-side-info.md

```diff
@@ -0,0 +1,173 @@
+    * [Import Functions to be Used as Client Function Dependencies](#import-functions-to-be-used-as-client-function-dependencies)
+* [Call Client Functions from Node.js Callbacks](#call-client-functions-from-nodejs-callbacks)
+* [Client Function Limitations](#client-function-limitations)
+
+## Client Function Constructor
+
+You can use the [ClientFunction](../../reference/test-api/global/clientfunction.md) constructor to create a client function.
+
+```js
+import { ClientFunction } from 'testcafe';
+const getWindowLocation = ClientFunction(() => window.location);
+```
+
+> Client functions cannot return DOM nodes. Use [selectors](select-page-elements.md) to achieve this.
+
+## Run Asynchronous Client Code
+
+TestCafe allows you to create client functions that run asynchronous code.
+
+You can pass a function that returns a Promise to the `ClientFunction` constructor to achieve this. In this instance, the client function will complete only when the Promise resolves.
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const performAsyncOperation = ClientFunction(() => {
+    return new Promise(resolve => {
+        window.setTimeout(resolve, 500); // some async operations
+    });
+});
+```
+
+## Execute Client Functions
+
+Call the client function with the `await` keyword to execute it.
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const getWindowLocation = ClientFunction(() => window.location);
+
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    const location = await getWindowLocation();
+});
+```
+
+## Overwrite Client Function Option
+
+You can overwrite client function options with the ClientFunction's [with method](../../reference/test-api/clientfunction/with.md).
+
+```js
+const cfWithDependency = cfWithoutDependency.with({
+    dependencies: { foo: 'bar' }
+});
+```
+
+## One-Time Client Code Execution
+
+You can use the [eval method](../../reference/test-api/testcontroller/eval.md) of the [test controller](../../reference/test-api/testcontroller/README.md) to create a client function and execute it immediately without saving it.
+
+```js
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    const docURI = await t.eval(() => document.documentURI);
+});
+```
+
+## Import Functions to be Used as Client Function Dependencies
+
+Assume you have a JS file `utils.js` with a function you need to use as a client function dependency in your test file.
+
+**utils.js**
+
+```js
+export function getDocumentURI() {
+    return document.documentURI;
+}
+```
+
+Note that TestCafe processes test files with [Babel](https://babeljs.io) internally. To avoid issues caused by code transpiling, use the [require](https://nodejs.org/api/modules.html#modules_require_id) function instead of the `import` statement to import client function dependencies.
```

Comment:
Change "To avoid issues caused by code transpiling" to "To avoid code transpiling issues"